recipes/pcapplusplus/all/conanfile.py

```diff
@@ -70,55 +72,68 @@ def _configure_sh_script(self):
+            replace_in_file(self, os.path.join(self.source_folder, "PcapPlusPlus.mk.common"),
                                   "-fPIC", "")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            files.patch(self, **patch)
+        apply_conandata_patches(self)
+        if self.settings.os != "Windows":
+            rename(self, os.path.join(self.source_folder, self._configure_sh_script), os.path.join(self.source_folder, "configure"))
+        else:
+            props_file = os.path.join(self.generators_folder, "conantoolchain.props")
+            vcxproj_templates = glob.glob(f"{self.source_folder}/mk/vs/*.vcxproj.template")
+            for template_file in vcxproj_templates:
+                # Load conan-generated conantoolchain.props before Microsoft.Cpp.props 
+                # so that we have precedence over PlatformToolset version
+                replace_in_file(self, template_file, '<Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />', 
+                                f'<Import Project="{props_file}" />\n<Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />')
+
+    def generate(self):
+        if self.settings.os == "Windows":
+            tc = MSBuildToolchain(self)
+            tc.generate()
+        else:
+            tc = AutotoolsToolchain(self)
+            tc.configure_args.clear() # this configure script does not accept the typical arguments
+            tc.configure_args.extend([
+                "--libpcap-include-dir", unix_path(self, self.dependencies["libpcap"].cpp_info.aggregated_components().includedirs[0]),
+                "--libpcap-lib-dir", unix_path(self, self.dependencies["libpcap"].cpp_info.aggregated_components().includedirs[0]),
+            ])
+
+            if self.options.immediate_mode:
+                tc.configure_args.append("--use-immediate-mode")
+            if is_apple_os(self) and "arm" in self.settings.arch:
+                tc.configure_args.append("--arm64")
+            
+            tc.generate()
 
     def build(self):
         self._patch_sources()
         if self.settings.os == "Windows":
             self._build_windows()
         else:
-            self._build_posix()
+            with chdir(self, self.source_folder):
+                autotools = Autotools(self)
+                autotools.configure()
+                autotools.make(target="libs")
 
     def _build_windows(self):
-        with files.chdir(self, self._source_subfolder):
+        with chdir(self, self.source_folder):
             config_args = [
                 "configure-windows-visual-studio.bat",
-                "--pcap-sdk", self.deps_cpp_info["npcap"].rootpath,
-                "--vs-version", "vs2015",
+                "--pcap-sdk", self.dependencies["npcap"].package_folder,
+                "--vs-version", "vs2015", # this will be later overridden by the props file generated by MSBuildToolchain
```

Comment:
Maybe we should update it to use VS 2019?
`"--vs-version",  "vs2019"`

Reply:
it's possible! In this case `vs2015` is passed simply because the script needs _a_ value, but it takes no practical effect. In practice, by loading (with priority) the conan toolchain props file generated by `MSBuildToolchain`, we are building for/with the toolset specified by the `compiler` Conan setting - this should be correct.  I can see how this looks misleading here though, hopefully the comment helps to clarify.

Reply:
@jcar87 I'm not sure I understand... PcapPlusPlus has a custom build system that creates VS project and solution files per VS version: https://pcapplusplus.github.io/docs/install/vs

I'm not sure how `MSBuildToolchain` works, but if you pass different values to `--vs-version` you'll get different solutions/projects.

This is something we're changing BTW and the next release will use CMake, which will make everything simpler and more streamlined. But until then it is what it it 😄